src/main/java/org/apache/ibatis/builder/annotation/MapperAnnotationBuilder.java

```diff
@@ -229,10 +229,12 @@ private void parseCacheRef() {
   private String parseResultMap(Method method) {
     Class<?> returnType = getReturnType(method);
     ConstructorArgs args = method.getAnnotation(ConstructorArgs.class);
+    Arg arg = method.getAnnotation(Arg.class);
     Results results = method.getAnnotation(Results.class);
+    Result result = method.getAnnotation(Result.class);
     TypeDiscriminator typeDiscriminator = method.getAnnotation(TypeDiscriminator.class);
     String resultMapId = generateResultMapName(method);
-    applyResultMap(resultMapId, returnType, argsIf(args), resultsIf(results), typeDiscriminator);
+    applyResultMap(resultMapId, returnType, argsIf(args, arg, method), resultsIf(results, result, method), typeDiscriminator);
```

Comment:
We can remove `resultsIf()` and `argsIf()` and use `method.getAnnotationsByType(Result.class)` and `method.getAnnotationsByType(Arg.class)` instead.

If we applied this change, the following tests fail.

- shouldErrorWhenSpecifyBothArgAndConstructorArgs
- shouldErrorWhenSpecifyBothResultAndResults

Those usages look weird, for sure, but we don't have to throw exception if we can handle them properly, I think.

Reply:
@harawata 

Thanks for your great advice. I've applied your suggestion and changed assertions for when  both annotations(`@Results` and `@Result` , `@ConstructorArgs` and `@Arg`) are specified.